kasa/protocols/smartprotocol.py

```diff
@@ -183,18 +185,18 @@ async def _execute_multiple_query(self, requests: dict, retry_count: int) -> dic
         multi_result: dict[str, Any] = {}
         smart_method = "multipleRequest"
 
+        end = len(requests)
+        # The SmartCamProtocol sends requests with a length 1 as a
+        # multipleRequest. The SmartProtocol doesn't so will never
+        # raise_on_error
+        raise_on_error = end == 1
+
```

Comment:
With `FORCE_SINGLE_REQUEST` this needs to happen before building the `multi_requests`

Reply:
Hmm, but why, when neither end nor raise_on_error is used in the comprehension?

Reply:
`raise_on_error` is used to ensure that single requests always raise an error which is the default behaviour. When we introduced `FORCE_SINGLE` I missed the fact that they will not error correctly because ` end = len(multi_requests)` gives zero.